### PR TITLE
adding endpoints to track logins and unique logins

### PR DESCRIPTION
**Pull request by joelhanson**

### CODE REVIEW

File: talentmap_api/stats/urls/logins.py [hunk 1]
```diff
@@ -0,0 +1,10 @@
+from django.conf.urls import url
+
+from talentmap_api.common.urls import get_list
+from talentmap_api.stats import views
+
+urlpatterns = [
+    url(r'^login/$', views.UserLoginActionView.as_view({'get': 'submit'}), name='stats.UserLogin-actions-submit'),
```
Comment (mjoyce91) [inline talentmap_api/stats/urls/logins.py]: Does this make more sense as a POST?

File: talentmap_api/stats/views.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+import logging
+logger = logging.getLogger(__name__)
+
+
+class UserLoginActionView(GenericViewSet):
+    '''
+    Tracks login for user
+    '''
+
+    permission_classes = (IsAuthenticated, isDjangoGroupMember('superuser'))
```
Comment (mjoyce91) [inline talentmap_api/stats/views.py]: Remove `isDjangoGroupMember('superuser')`. Right now only superusers can hit this endpoint

File: talentmap_api/stats/views.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    def submit(self, request, format=None):
+        '''
+        Logs a user login
+
+        Returns 204 if the action is a success
+        '''
+        user = UserProfile.objects.get(user=self.request.user)
+        login_instance = LoginInstance()
+
+        logger.info(f"User {self.request.user.id}:{self.request.user} is logging in")
```
Comment (mjoyce91) [inline talentmap_api/stats/views.py]: Where does this get logged to?

File: talentmap_api/stats/views.py (hunk 2 above)
Comment (joelhanson) [reply talentmap_api/stats/views.py]: Isn't that the point? This is supposed to be just for admins, is it not?

File: talentmap_api/stats/views.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+        login_instance.user = user
+        login_instance.date_of_login = datetime.datetime.now()
+        login_instance.save()
+        # status=status.HTTP_204_NO_CONTENT,
+        # data = 'It worked'
```
Comment (mjoyce91) [inline talentmap_api/stats/views.py]: Comments needed?

Comment (joelhanson) [reply talentmap_api/stats/views.py]: Need to be removed. Missed this before the commit.

File: talentmap_api/stats/urls/logins.py (hunk 1 above)
Comment (joelhanson) [reply talentmap_api/stats/urls/logins.py]: Probably. I'll change it

**Comment by mjoyce91** [comment]
This is nice!

Screen Shot 2019-08-20 at 10 05 26 AM

File: talentmap_api/stats/views.py (hunk 2 above)
Comment (mjoyce91) [reply talentmap_api/stats/views.py]: Yeah but this one specifically is the endpoint to "submit" a login

Comment (joelhanson) [reply talentmap_api/stats/views.py]: As always, you raise nothing but good points. I'll update

**Comment by mjoyce91** [comment]
Looks good, failing tests though

**Comment by joelhanson** [comment]
@mjoyce91 I noticed that, too. Not sure why, those tests are unrelated to my code. Looking into it now.

**Pull request merged**